File: Linked List/Circular_Linked_list.c
//Circular Linked List

#include<stdio.h>
#include<stdlib.h>


typedef struct node{
    int info;
    struct node* link;
}Node;

//Insert to empty list 
void display(Node *last){
    if(last == NULL){
        printf("List is empty");
        return;
    }
    Node *p;
    p = last->link;
    do{
        printf("%d ",p->info);
        p = p->link;
    }while(p!= last->link);
    printf("\n");
    return;
}



Node * add_to_empty(Node *last,int data){
    Node * tmp = (Node*)malloc(sizeof(Node));
    tmp->info = data;
    last = tmp;
    tmp->link = last;
    return last;
}



Node* add_to_beg(Node* last,int data){
    Node *tmp = (Node*)malloc(sizeof(Node));
    tmp->info = data;
    tmp->link = last->link;
    last->link= tmp;
    return last;
}

int main(void){
    Node* last = NULL;
    last = add_to_empty(last,1);
    last = add_to_beg(last,2);
    display(last);
    return 0;
}

File: Linked List/Java/CircularLinkedList.java
class CircularLinkedList{
    class Node{
        int data;
        Node next;
        public Node(int data){
            this.data = data;
        }
    }
    Node last = null;
    int length = 0;
    public void print(){
        Node p;
        if(last == null){
            System.out.println("List is empty");
            return;
        }
        p = last.next;
        do{
            System.out.println(p.data);
            p = p.next;
        }while(p != last.next);
        return;
    }
    public void insert(int data){
        Node p;
        Node tmp = new Node(data);
        if(last == null){
            last = tmp;
            last.next = tmp;
            length++;
            return;
        }
        tmp.next = last.next;
        last.next = tmp;
        last = tmp;
        length++;
        return;
    }
}

public class Main{
    public static void main(String args[]){
        CircularLinkedList c = new CircularLinkedList();
        c.insert(1);
        c.insert(3);
        c.print();
        
    }
}
